### PR TITLE
Toaster component

### PR DESCRIPTION
**Pull request by xdrdak**
## Description

This PR adds a new toaster component. It uses the [react-toast-notifications](https://github.com/jossmac/react-toast-notifications) library for pretty much most of the heavy lifting.

We've created 2 components that are auto-hooked into an extended `<ToastProvider />` component, meaning everything should be configured to the DSD defaults, but we still allow changes (any props that should work with the regular ToastProvider will work with ours)

I've also taken the time to update react-testing-library to its latest version to take advantage of the `screen` function.

## How to test?

- Checkout branch, run `yarn dev`
- Open [Storybook](http://localhost:6006)
- Or check the deploy preview on Netlify (link available in comments)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/lightspeed/flame/blob/master/.github/CONTRIBUTING.md) guide
- [x] I have prepared [CHANGELOGs](https://github.com/lightspeed/flame/blob/master/.github/CONTRIBUTING.md#git-workflow) for release
- [x] I have tested my changes on [supported browsers](https://browserl.ist/?q=%3E0.25%25%2C+not+op_mini+all%2C+not+ie+%3C%3D+11)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/flame/src/Toaster/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import React from 'react';
+import { FlameTheme, FlameGlobalStyles } from '@lightspeed/flame/Core';
+import { ToasterProvider } from '@lightspeed/flame/Toaster';
+
+const App = () => (
+  <FlameTheme>
+    <FlameGlobalStyles />
+    <ToasterProvider>
+      <div>{/* The rest of your app */}</div>
+      <div>{/* The rest of your app */}</div>
```
Comment (glambert) [inline packages/flame/src/Toaster/README.md:22]: Do we need the same example twice here?

File: packages/flame/src/Toaster/story.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import { storiesOf } from '@storybook/react';
+import { withReadme } from 'storybook-readme';
+import Readme from './README.md';
+import { ToasterProvider, useToasts } from './index';
+import { ActionableToastContent, Toaster } from './Toaster';
+import { Button } from '../Button';
+import { Heading2 } from '../Text';
```
Comment (glambert) [inline packages/flame/src/Toaster/story.tsx:8]:
```suggestion
import { Button } from '../Button';
import { Heading2 } from '../Text';
import Readme from './README.md';
import { ActionableToastContent, Toaster } from './Toaster';
import { ToasterProvider, useToasts } from './index';
```

Nit to go from absolute to (closest) relative imports. We could enable an eslint rule for this if we want to automate: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md

File: packages/flame/src/Toaster/story.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  return (
+    <Spacer>
+      <Button
+        type="button"
+        onClick={() => {
+          addActionableToast({
+            content: `I have an action!`,
+            actionTitle: 'Undo',
+            actionCallback: () => {
+              console.log('You clicked the undo button!');
```
Comment (glambert) [inline packages/flame/src/Toaster/story.tsx:61]:
```suggestion
              // eslint-disable-next-line no-console
              console.log('You clicked the undo button!');
```

File: packages/flame/src/Toaster/story.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+            },
+          });
+        }}
+      >
+        Generate success
+      </Button>
```
Comment (glambert) [inline packages/flame/src/Toaster/story.tsx:67]: Non-blocking: should we customize the default active/focus state for the close button or leave as is? This is with latest Chrome:

image

File: packages/flame/src/Toaster/story.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      <Button
+        type="button"
+        onClick={() => {
+          addActionableToast(
+            {
+              content: `I have an action!`,
+              actionTitle: 'Undo',
+              actionCallback: () => {
+                console.log('You clicked the undo button!');
```
Comment (glambert) [inline packages/flame/src/Toaster/story.tsx:76]:
```suggestion
                // eslint-disable-next-line no-console
                console.log('You clicked the undo button!');
```

File: packages/flame/src/Toaster/Toaster.tsx [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+import { IconDanger } from '../Icon/Danger';
+import { IconMathMultiply } from '../Icon/MathMultiply';
+import { Box, Flex } from '../Core';
+
+const toastStates = {
+  entering: { transform: 'translate3d(0, 120%, 0)', opacity: 0 },
+  entered: { transform: 'translate3d(0,0,0)', opacity: 1 },
+  exiting: { transform: 'translate3d(0, 120%, 0)', opacity: 0 },
+  exited: { transform: 'translate3d(0, 120%, 0)', opacity: 0 },
+};
```
Comment (glambert) [inline packages/flame/src/Toaster/Toaster.tsx:22]: Was multi-stacking animation implemented from specs? There is a slight "jump" after a an extra toaster is shown:

Kapture 2020-06-08 at 15 20 31

File: packages/flame/src/Toaster/story.tsx (hunk 2 above)
Comment (xdrdak) [reply packages/flame/src/Toaster/story.tsx:8]: open a pr to the LS linter if you feel strongly about it

File: packages/flame/src/Toaster/Toaster.tsx (hunk 6 above)
Comment (xdrdak) [reply packages/flame/src/Toaster/Toaster.tsx:22]: Multi stacking is apparently allowed, but no specs are written.

If anything, users can disable the functionality very easily by checking if there are any notifications in the stack.

jump is due to the underlying lib not properly calculating where things should pop, I would need to PR or hack around the issue. Its not as jarring as their own example luckily.

File: packages/flame/src/Toaster/story.tsx (hunk 2 above)
Comment (glambert) [reply packages/flame/src/Toaster/story.tsx:8]: Meh, this is what we currently have based on the version of the `airbnb` setting we extend:

Comment (glambert) [reply packages/flame/src/Toaster/story.tsx:8]: We could go with the latest version of [`eslint-plugin-import`'s order](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md#groups-array):

>The default value is ["builtin", "external", "parent", "sibling", "index"].

File: packages/flame/src/Toaster/story.tsx (hunk 4 above)
Comment (xdrdak) [reply packages/flame/src/Toaster/story.tsx:67]: @maartenafink

File: packages/flame/src/Toaster/story.tsx (hunk 2 above)
Comment (xdrdak) [reply packages/flame/src/Toaster/story.tsx:8]: we could add it, but that should be done in a different PR.

Comment (xdrdak) [reply packages/flame/src/Toaster/story.tsx:8]: tracking here #91

File: packages/flame/src/Toaster/story.tsx [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+        <Toaster appearance="error" {...commonProps} autoDismiss>
+          <ActionableToastContent actionCallback={noop} actionTitle="Undo">
+            This is a success toaster with action
+          </ActionableToastContent>
+        </Toaster>
+      </div>
+    </ToasterProvider>
+  );
+});
```
Comment (glambert) [inline packages/flame/src/Toaster/story.tsx:179]: Is it possible to capture a Toaster with a (static) progress state for Percy?

Comment (xdrdak) [reply packages/flame/src/Toaster/story.tsx:179]: Not unless I remove animations entirely from the component and change a chunk of the underlying implementation just to have visual diffing working.

Not worth it.

Comment (glambert) [reply packages/flame/src/Toaster/story.tsx:179]: Giving it a spin to see [how Percy behaves](https://docs.percy.io/docs/animations#automatic-animation-freezing-in-percy) by enabling progress on the actionable examples.

Comment (glambert) [reply packages/flame/src/Toaster/story.tsx:179]: No dice - oh well, let's keep as is for now :+1:

**Pull request merged**